six/modules/c++/six.sicd/include/six/sicd/ComplexData.h

```diff
@@ -275,16 +276,17 @@ class ComplexData: public Data
      *  differently for SICD than SIDD.  Any profile parameters for
      *  FileSecurity that do not conform to Classification transfer must
      *  be initialized directly.
-     *  Maps to SICD/CollectionInfo/Classification
      */
     virtual const Classification& getClassification() const
     {
-        return collectionInformation->getClassification();
+        mClassification.level = collectionInformation->getClassificationLevel();
+        return mClassification;
     }
 
     virtual Classification& getClassification()
```

Comment:
Why were we overloading this function in the first place?

Reply:
You mean specifically the non-const accessor? Clients can go through it to change security options in the NITF header (see NITFReadControl). I'd rather not expose this and just have a 'setSecurityOption` method instead, but then I'd rather not break more code than I have to.